### DISCUSSION TITLE
VAAPI seemingly not enabled on docker-compose ffmpeg

### DISCUSSION THREAD

**Discussion by MutantPiggieGolem1** [photoprism/photoprism#4676] [Q&A]
I have an AMD Radeon HD 8670D, which should support h264 encode with vaapi. I have this setting enabled in the config, but nevertheless always get `h264_vaapi: failed to transcode` in the logs. I have this section in my docker-compose.yml.
```## Video Transcoding (https://docs.photoprism.app/getting-started/advanced/transcoding/):
      PHOTOPRISM_FFMPEG_ENCODER: "vaapi"             # H.264/AVC encoder (software, intel, nvidia, apple, raspberry, or vaapi)
      VAAPI_MPEG4_ENABLED: 1
      PHOTOPRISM_FFMPEG_SIZE: "3840"                 # video size limit in pixels (720-7680) (default: 3840)
      # PHOTOPRISM_FFMPEG_BITRATE: "32"              # video bitrate limit in Mbit/s (default: 50)
      ## Run/install on first startup (options: update https gpu ffmpeg tensorflow davfs clitools clean):
      PHOTOPRISM_INIT: "update gpu ffmpeg tensorflow"
```
And running ffmpeg inside the container shows that it was built with these flags:
```ffmpeg version 7.0.2-static https://johnvansickle.com/ffmpeg/  Copyright (c) 2000-2024 the FFmpeg developers
  built with gcc 8 (Debian 8.3.0-6)
  configuration: --enable-gpl --enable-version3 --enable-static --disable-debug --disable-ffplay --disable-indev=sndio --disable-outdev=sndio --cc=gcc --enable-fontconfig --enable-frei0r --enable-gnutls --enable-gmp --enable-libgme --enable-gray --enable-libaom --enable-libfribidi --enable-libass --enable-libvmaf --enable-libfreetype --enable-libmp3lame --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libopenjpeg --enable-librubberband --enable-libsoxr --enable-libspeex --enable-libsrt --enable-libvorbis --enable-libopus --enable-libtheora --enable-libvidstab --enable-libvo-amrwbenc --enable-libvpx --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxml2 --enable-libdav1d --enable-libxvid --enable-libzvbi --enable-libzimg
  libavutil      59.  8.100 / 59.  8.100
  libavcodec     61.  3.100 / 61.  3.100
  libavformat    61.  1.100 / 61.  1.100
  libavdevice    61.  1.100 / 61.  1.100
  libavfilter    10.  1.100 / 10.  1.100
  libswscale      8.  1.100 /  8.  1.100
  libswresample   5.  1.100 /  5.  1.100
  libpostproc    58.  1.100 / 58.  1.100
```

Why has ffmpeg been built without `--enable-vaapi`? Or why is the vaapi encoder (`h264_vaapi`) not present in the list produced by `ffmpeg -encoders`?

**Comment by lastzero** [comment]
You can either install a newer version statically, or use the pre-installed distribution package to enable hardware transcoding, which (in most cases) requires dynamic linking against the driver libraries and/or a special license to avoid GPL conflicts:
- https://docs.photoprism.app/getting-started/advanced/transcoding/#software-transcoding

**Reply by MutantPiggieGolem1** [reply]
I apologize for forgetting to mention that I am using the official docker-compose, straight from the website.
I believe the hardware acceleration distribution package is already installed on that image (the guide says it should be). /dev/dri/ is already linked into the container as a device. I am not sure how to set up the special license, but the `mesa-va-drivers` package, which provides support for my card, should fully comply with GPL.

This Emby PR may be related: https://github.com/linuxserver/docker-baseimage-ubuntu/issues/44

**Reply by lastzero** [reply]
See the link I shared with you: `PHOTOPRISM_INIT: "ffmpeg"` will install the static build available at https://johnvansickle.com/ffmpeg/. If you want/need to use FFmpeg 7 with hardware acceleration, you can upgrade to our ":preview" image which is based on the latest Ubuntu base image. In any case, you must not specify "ffmpeg" as init action.

**Comment by pmferraro** [comment]
Can you use your AMD card with ffmpeg, i have a radeon 6900xt and cant activate vaapi

**Reply by lastzero** [reply]
We haven't tested it yet. However, since our image is based on a standard Ubuntu Linux, you can follow their documentation or ask their community what is needed to get hardware transcoding working with your GPU/CPU.

**As mentioned above, you must use the version of FFmpeg that comes with the container image, as the static build you can install with `PHOTOPRISM_INIT:"ffmpeg"` does NOT support any form of GPU acceleration.**
